tracing-appender/src/rolling.rs

```diff
@@ -558,17 +562,68 @@ impl Inner {
                     .unwrap_or(0),
             ),
             rotation,
+            max_files,
         };
         Ok((inner, writer))
     }
 
+    fn prune_old_logs(&self, max_files: usize) {
+        let files = fs::read_dir(&self.log_directory).map(|dir| {
+            dir.filter_map(Result::ok)
+                .filter(|entry| {
+                    entry.file_name().to_string_lossy().contains(
+                        &self
+                            .log_filename_prefix
+                            .clone()
+                            .expect("prefix is always present"),
+                    )
+                })
```

Comment:
`.log_filename_prefix` is `Option<String>`.
Without `clone()`, the borrow checker is not happy:
```
move occurs because `self.log_filename_prefix` has type `Option<String>`, which does not implement the `Copy` trait
```

Reply:
I think you are completely right on your first comment. When I wrote the code, I wrote it for my specific needs and forgot to change this part to general purpose. 
Your suggestions sounds pretty reasonable to me, implementing now 💯

Reply:
Adding the check of parsing the filename as a date required me to add some previous stuff back (adding Date and Format related things). If there is better way doing that, I'm open to suggestions. 

If not, you may want to reintroduce the `parsing` feature again maybe?